syntax = "proto3";

package stt;

import "google/protobuf/empty.proto";

service STTService {
  rpc RecognizeStream(stream STTStreamRequest) returns (google.protobuf.Empty);
}

message STTStreamRequest {
  oneof request_data {
    RecognitionConfig config = 1;
    bytes audio_chunk = 2;
  }
}

message RecognitionConfig {
  string session_id = 1;          // STT 서비스 내부에서 사용할 수 있는 세션 ID (선택적)
  string language = 2;
  string frontend_session_id = 3; //
}